### PR TITLE
nft test-cases : fixed and updated

### PR DESCRIPTION
**Pull request by githubckgoh1439**
1. update nft test cases.
2. checking move to first layer checking at validation.go
3. fix some nft test cases.

### CODE REVIEW

File: app/validation.go [hunk 1]
```diff
@@ -289,20 +317,49 @@ func (app *mxwApp) validateMsg(ctx sdkTypes.Context, msg sdkTypes.Msg) sdkTypes.
...
 			return types.ErrTokenFrozen()
 		}
-	case nonFungible.MsgBurnNonFungibleToken:
+
+		//1. [Mint (by Public==TRUE) non fungible token(TNFT-public-01) - Error, Public token can only be minted to itself.]
+		var token = new(nonFungible.Token)
+		app.nonFungibleTokenKeeper.GetTokenDataInfo(ctx, msg.Symbol, token)
+		if token.Flags.HasFlag(0x0080) {
+			OwnerAcc := msg.Owner
+			NewOwnerAcc := msg.To
```
Comment (yenkhoon) [inline app/validation.go]: variable name always start with small letter.

Comment (githubckgoh1439) [reply app/validation.go]: fixed

File: app/validation.go [hunk 2]
```diff
@@ -182,6 +182,14 @@ func (app *mxwApp) validateMsg(ctx sdkTypes.Context, msg sdkTypes.Msg) sdkTypes.
 		if app.tokenKeeper.IsTokenFrozen(ctx, msg.Symbol) {
 			return types.ErrTokenFrozen()
 		}
+
```
Comment (b00f) [inline app/validation.go]: 1- here is for fungible token not nin-fungible token.
2- please create a function for check token existence.

File: app/validation.go [hunk 3]
```diff
@@ -269,6 +277,25 @@ func (app *mxwApp) validateMsg(ctx sdkTypes.Context, msg sdkTypes.Msg) sdkTypes.
 			}
 		}
 	case nonFungible.MsgSetNonFungibleItemStatus:
+		var token = new(nonFungible.Token)
+		if exists := app.nonFungibleTokenKeeper.GetTokenDataInfo(ctx, msg.ItemPayload.Item.Symbol, token); !exists {
```
Comment (b00f) [inline app/validation.go]: 1- add IsTokenExists function
2- add IsTokenItemExists function

File: app/validation.go [hunk 4]
```diff
@@ -289,20 +317,49 @@ func (app *mxwApp) validateMsg(ctx sdkTypes.Context, msg sdkTypes.Msg) sdkTypes.
...
+	case nonFungible.MsgMintNonFungibleItem:
 		if !app.nonFungibleTokenKeeper.CheckApprovedToken(ctx, msg.Symbol) {
 			return types.ErrTokenInvalid()
 		}
 		if app.nonFungibleTokenKeeper.IsTokenFrozen(ctx, msg.Symbol) {
 			return types.ErrTokenFrozen()
 		}
-	case nonFungible.MsgBurnNonFungibleToken:
+
+		//1. [Mint (by Public==TRUE) non fungible token(TNFT-public-01) - Error, Public token can only be minted to itself.]
```
Comment (b00f) [inline app/validation.go]: Explain more about this comment.

File: app/validation.go [hunk 5]
```diff
@@ -289,20 +317,49 @@ func (app *mxwApp) validateMsg(ctx sdkTypes.Context, msg sdkTypes.Msg) sdkTypes.
...
+		var token = new(nonFungible.Token)
+		app.nonFungibleTokenKeeper.GetTokenDataInfo(ctx, msg.Symbol, token)
+		if token.Flags.HasFlag(0x0080) {
```
Comment (b00f) [inline app/validation.go]: Create function for this flag

**Pull request merged**